File: <100/085/085.cpp
#include<iostream>
using namespace std;


int area(int m, int n){
  int tn = n * (n+1) / 2;
  int tm = m * (m+1) / 2;
  return tn * tm;
}


int main(){
  int minimum = 1e9, target = 2e6;
  int answer;

  for(int m = 1; m <= 200; ++m)for(int n = m; n <= 200; ++n){
    int a = area(m, n);
    int gap = abs(a - target);
    if(gap < minimum){
      minimum = gap;
      answer = m * n;
    }
  }

  cout << answer << endl;
  return 0;
}

File: <100/085/085.py
def area(m: int, n: int) -> int:
    tn = n * (n+1) // 2
    tm = m * (m+1) // 2
    return tn * tm


def main() -> int:
    minimum = 10**9
    target = 2 * 10**6
    answer = 1000000
    for m in range(1, 201):
        for n in range(m, 201):
            a = area(m, n)
            gap = abs(target - a)
            if gap < minimum:
                minimum = gap
                answer = m * n

    return answer


if __name__ == "__main__":
    answer = main()
    print(answer)
